File: src/main/java/io/github/aphelium/modules/coin/entities/Crypto.java
package io.github.aphelium.modules.coin.entities;

import io.github.aphelium.modules.brokers.entities.Broker;
import lombok.Data;

import javax.persistence.*;
import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.Set;
import java.util.UUID;

@Entity
@Data
@Table(name = "TB_CRYPTO")
public class Crypto implements Serializable {

    @Serial
    private static final long serialVersionUID = -8080540494839892473L;

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private UUID id;

    @Column(nullable = false, unique = true, length = 50)
    private String name;

    @Column(nullable = true)
    private String description;

    @Column(nullable = false, unique = true)
    private String code;

    @Column
    private Boolean isActive;

    @Column(nullable = true)
    private String logo;

    @Column(nullable = false)
    private String firstHistoricalDate;

    @Column(nullable = true)
    private String lastHistoricalDate;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

}

File: src/main/kotlin/io/github/aphelium/modules/coin/entities/Crypto.kt
package io.github.aphelium.modules.coin.entities

import java.io.Serial
import java.io.Serializable
import java.time.LocalDateTime
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "TB_CRYPTO")
class Crypto : Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    val id: UUID? = null

    @Column(nullable = false, unique = true, length = 50)
    val name: String? = null

    @Column(nullable = true)
    val description: String? = null

    @Column(nullable = false, unique = true)
    val code: String? = null

    @Column
    val isActive: Boolean? = null

    @Column(nullable = true)
    val logo: String? = null

    @Column(nullable = false)
    val firstHistoricalDate: String? = null

    @Column(nullable = true)
    val lastHistoricalDate: String? = null

    @Column(nullable = false)
    var createdAt: LocalDateTime? = null

    @Column(nullable = false)
    var updatedAt: LocalDateTime? = null

    companion object {
        @Serial
        private val serialVersionUID = -8080540494839892473L
    }
}